src/d/actor/d_a_npc_bs1.cpp

```diff
@@ -14,56 +150,269 @@ daNpc_Bs1_childHIO_c::daNpc_Bs1_childHIO_c() {
+                offset.x = 0.0f;
+                offset.y = 0.0f;
+                offset.z = 0.0f;
+                mDoMtx_XrotM(*calc_mtx, i_this->getHead_y());
+                mDoMtx_ZrotM(*calc_mtx, -i_this->getHead_x());
+                MtxPosition(&offset, &l_attention);
+                i_this->setAttentionBasePos(l_attention);
+                offset.x = 28.0f;
+                offset.y = 20.0f;
+                offset.z = 0.0f;
+                // cXyz other;
+                MtxPosition(&offset, &l_attention);
+                i_this->setEyePos(l_attention);
+                if (i_this->m72F != (u8)-1) {
+                    i_this->m72F += 1;
+                }
+            } else if (jntNo == (s8)i_this->getBackboneJntNum()) {
+                mDoMtx_XrotM(*calc_mtx, i_this->getBackbone_y());
+                mDoMtx_ZrotM(*calc_mtx, -i_this->getBackbone_x());
+            }
+            cMtx_copy(*calc_mtx, J3DSys::mCurrentMtx);
+            model->setAnmMtx(jntNo, *calc_mtx);
+        }
+    }
+    return TRUE;
 }
 
 /* 00000658-0000076C       .text initTexPatternAnm__11daNpc_Bs1_cFb */
-void daNpc_Bs1_c::initTexPatternAnm(bool) {
+BOOL daNpc_Bs1_c::initTexPatternAnm(bool i_modify) {
     /* Nonmatching */
+    J3DModelData* modelData = mpMorf->getModel()->getModelData();
+    m_head_tex_pattern = (J3DAnmTexPattern*)dComIfG_getObjectRes("Bs", l_btp_ix_tbl[m828]);
+    JUT_ASSERT(0x1bd, m_head_tex_pattern != 0);
+    if (!mBtpAnm.init(modelData, m_head_tex_pattern, 1, 2, 1.0f, 0, -1, i_modify, 0)) {
+        return FALSE;
+    }
+    i_frame = 0;
+    m2C6 = 0;
+    return TRUE;
 }
 
 /* 0000076C-000007F8       .text playTexPatternAnm__11daNpc_Bs1_cFv */
 void daNpc_Bs1_c::playTexPatternAnm() {
     /* Nonmatching */
+    if (cLib_calcTimer(&m2C6) == 0) {
+        s16 frameMax = m_head_tex_pattern->getFrameMax();
+        if (frameMax >= i_frame) {
+            i_frame -= frameMax;
+            m2C6 = cM_rndF(100.0f) + 30.0f;
+        } else {
+            i_frame += 1;
+        }
+    }
 }
 
 /* 000007F8-0000087C       .text setAnm__11daNpc_Bs1_cFSc */
-void daNpc_Bs1_c::setAnm(signed char) {
+void daNpc_Bs1_c::setAnm(signed char index) {
     /* Nonmatching */
+    if (index != m829 || m829 == -1) {
+        m829 = index;
+        dNpc_setAnm_2(mpMorf, play_mode_tbl[index], morf_frame_tbl[index], play_speed_tbl[index],
+                      l_bck_ix_tbl[index], -1, "Bs");
+    }
 }
 
 /* 0000087C-000008C0       .text setTexAnm__11daNpc_Bs1_cFSc */
-void daNpc_Bs1_c::setTexAnm(signed char) {
-    /* Nonmatching */
+void daNpc_Bs1_c::setTexAnm(signed char value) {
+    if (m828 != value || m828 == -1) {
+        m828 = value;
+        initTexPatternAnm(TRUE);
+    }
 }
 
 /* 000008C0-00000BD8       .text setAnmFromMsgTag__11daNpc_Bs1_cFv */
 void daNpc_Bs1_c::setAnmFromMsgTag() {
     /* Nonmatching */
+    switch (dComIfGp_getMesgAnimeAttrInfo()) {
+    case 0:
+        setAnm(ANM_UNK_0);
+        setTexAnm(ANM_UNK_1);
```

Comment:
anm indexes the bck list and texanm indexes the btp list, so they shouldn't be represented by the same enum since they're unrelated values.

also, there are a lot of places in this file where the enum isn't used consistently and raw integer literals are used.

might be easier to only use integer literals until the entire file is decompiled, then add the two enums in a documentation pass later